### PR TITLE
Task 7 McKucia, first implementation

### LINKED ISSUE #8: 7. Class Player
Napisz klasę Player, która posiadać będzie 3 pola:

std::shared_ptr<Ship> ship_ (dla chętnych, spróbuj użyć std::unique_ptr<>)
money_
availableSpace_
Dopisz także odpowiednie gettery oraz konstruktor.

Dopisz także 2 funkcje, które powinny zwracać dane ze statku:

size_t getSpeed() const
Cargo* getCargo(size_t index) const

### LINKED ISSUE #9: 8. Class Player - availableSpace
W klasie Player napisz prywatną funkcję, która obliczać będzie availableSpace_ na podstawie aktualnej ilości towaru na statku.

### PR DESCRIPTION
**Pull request by McKucia**
Player.hpp + .cpp

### CODE REVIEW

File: shm/Player.cpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#include "Player.hpp"
+
+#include <algorithm>
+
+Cargo* Player::getCargo(size_t index) const {
+    if (index > (ship_->getCargo().size()))
+        return nullptr;
```
Comment (wbrejecki) [inline shm/Player.cpp]: It's a good practice to use brackets in if statements even if there is only one line of instructions

File: shm/Player.cpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    return &ship_->getCargo()[index];
+}
+
+void Player::countAvailableSpace() {
+
+    auto amoutOfCargo = ship_->getCargo().size();
```
Comment (wbrejecki) [inline shm/Player.cpp:14]: I don't think this it is going to work. Class 'Cargo' has field called 'amount_' and you should take it into consideration. Correct me if I'm wrong.

File: shm/Player.cpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    auto availableSpaceTmp = ship_->getCapacity() - amoutOfCargo;
+
+    if(availableSpaceTmp <= 0)
+        availableSpace_ = availableSpaceTmp;
+    else
+        availableSpace_ = availableSpaceTmp;
```
Comment (wbrejecki) [inline shm/Player.cpp]: You can easily simplify it.

File: shm/Player.cpp (hunk 2 above)
Comment (McKucia) [reply shm/Player.cpp:14]: Sure but this function is working on specific object of type "Player". Each player has a ship_ field, and every ship_ has getCargo() function. So i'm working on this specific player - `auto amount of Cargo - this->ship_->getCargo().size(); `

Comment (McKucia) [reply shm/Player.cpp:14]: I don't know if I answered your question correctly but you can explain it more clearly on discord : D

File: shm/Player.cpp (hunk 3 above)
Comment (McKucia) [reply shm/Player.cpp]: any hint? ; >

File: shm/Player.cpp (hunk 2 above)
Comment (wbrejecki) [reply shm/Player.cpp:14]: Yea, but what type is function getCargo() in Cargo class? I think it should be std::vector<Cargo> getCargo() const. So we have a vector of Cargos and each Cargo has a field called amount_ which tells us how many pieces of Cargo there is. Statement: getCargo().size() tells us how many **types** of cargo we have but not exact **quantity**. At least that's the way I see it. I'm not saying I'm right for sure :)

File: shm/Player.cpp (hunk 3 above)
Comment (wbrejecki) [reply shm/Player.cpp]: First of all - brackets :D 
I think 'else' is unnecessary (The word 'else', not what is after it). Code would work the same without it. Also you don't have to create two new variables.

Comment (McKucia) [reply shm/Player.cpp]: Sure, it should be `0`, thx

File: shm/Player.cpp (hunk 2 above)
Comment (McKucia) [reply shm/Player.cpp:14]: function getCargo() is in ship class, that was my idea

**Review by McKucia — COMMENTED** [review]
New commit, pls approve or not.
Also pls add  `std::vector<Cargo>& getCargo()  { return cargo_; };`
and `std::vector<Cargo> cargo_;` in ship.hpp!!!!!!!!!!!!

File: shm/Player.cpp (hunk 2 above)
Comment (wbrejecki) [reply shm/Player.cpp:14]: BTW. Method countAvailableSpace() is task 8 so you also did task 8 :smile:

**Comment by wbrejecki** [comment]
> New commit, pls approve or not.
> Also pls add `std::vector<Cargo>& getCargo() { return cargo_; };`
> and `std::vector<Cargo> cargo_;` in ship.hpp!!!!!!!!!!!!

Done :+1:

**Review by wbrejecki — APPROVED** [review]

**Review by kali-03 — APPROVED** [review]

**Pull request merged**